cmd/pod-scaler/frontend/src/app/SearchTreeView/SearchTreeView.tsx

```diff
@@ -0,0 +1,232 @@
+import * as React from 'react';
+import {Alert, Spinner, TreeView, TreeViewDataItem} from '@patternfly/react-core';
+import {selectParents, selectTrees, updateNames, updateParents, updateTrees} from "@app/store";
+import {useAppDispatch, useAppSelector} from "@app/hook";
+
+export interface SearchTreeViewProps {
+  /** URL to fetch raw data from */
+  dataUrl: string;
+  /** A child ID to render as active at first */
+  activeChild?: string;
+  /** Set the active item to let others know */
+  setActiveItem: React.Dispatch<React.SetStateAction<TreeViewDataItem>>;
+}
+
+type Node = {
+  name: string;
+  field: string;
+  children?: Node[];
+};
+
+/** withParameter adds a parameter to the source set, doing a deep copy to not mutate the source */
+const withParameter = (parameters: URLSearchParams, name: string, field: string): URLSearchParams => {
+  const newRecord: URLSearchParams = new URLSearchParams();
+  parameters.forEach(function (value: string, key: string) {
+    newRecord.set(key, value);
+  })
+  newRecord.set(field, name);
+  return newRecord;
+};
+
+/** toId is a deterministic transformation from URL parameters to a string */
+export const toId = (parameters: URLSearchParams): string => {
+  parameters.sort();
+  return parameters.toString();
+}
+
+/** toAllItems converts the backend forest response into TreeViewDataItems */
+const toAllItems = (data: Node[]): [TreeViewDataItem[], Record<string, string>, Record<string, string>] => {
+  const parentsById: Record<string, string> = {};
+  const namesById: Record<string, string> = {};
+  const items: TreeViewDataItem[] = [];
+  const parameters: URLSearchParams = new URLSearchParams();
+  for (const datum of data) {
+    items.push(toItems(datum, parameters, parentsById, namesById))
+  }
```

Comment:
Consider using `Array.map`

Reply:
Too fancy for my small brain :(

How can you tell I write Go?